app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,47 @@
+class SitemapController < ApplicationController
+
+  def sitemap
+    #if false
+      #redirect_to "/" # The CDN then fetches the sitemap.xml from the sitemap/generate URL which points to the `generate` action
+    #else
+    render_site_map
+    #end
+  end
+
+  def generate
+    render_site_map
+  end
+
+  private
+
+    def render_site_map
+      respond_to do |format|
+
+        data = Rails.cache.fetch(@current_community, expires_in: 24.hours) do
+          adapter = SitemapGenerator::NeverWriteAdapter.new
+
+          if @current_community.use_domain 
+            default_host = "https://"+@current_community.domain
+          else 
+            default_host = "https://"+APP_CONFIG.domain
+          end
+
+          community_id = @current_community.id
+
+          SitemapGenerator::Sitemap.create(
+                :default_host => default_host,
+                :verbose => false,
+                :adapter => adapter) do
+                  Listing.where(deleted: false, open: true, community_id: community_id).find_each do |listing|
```

Comment:
A couple of things.

I realized that we use the id rather than the object, which means that the sitemap points to `/listings/42` as opposed to `/listings/42-cool-listing-name`.  The latter is obviously superior, so I'll see if I can fix that.  (Not sure what that means for the `pluck` suggestion.)

I can definitely add a wrapper method in the controller to abstract the activemodel code.  

Re: large number of links, I think we should limit to at most 50k (see https://github.com/kjvarga/sitemap_generator/blob/master/lib/sitemap_generator.rb#L26 ).  Otherwise we get into a sitemap situation where we have to have multiple sitemap files, and I haven't tested that.  As for whether it should be less, that's really a question for you.  I'd say that if I were running a marketplace, I'd want all the listings to be searched, and would rather deal with the performance issues.  

Re: performance--it sounds like using `currently_open` is OK for now?  I'll implement that.